File: python/meteroid/models/connected_account.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .connected_account_id import ConnectedAccountId
from .connection_status import ConnectionStatus
from .connection_type import ConnectionType
from .country_code import CountryCode
from .customer_id import CustomerId
from .onboarding_mode import OnboardingMode
from .organization_id import OrganizationId
from .tenant_id import TenantId


@dataclasses.dataclass
class ConnectedAccount(BaseModel):
    """A connected account (relationship between platform and connected org)"""

    connection_type: ConnectionType

    created_at: datetime

    id: ConnectedAccountId

    onboarding_mode: OnboardingMode

    platform_organization_id: OrganizationId

    status: ConnectionStatus

    connected_organization_id: t.Optional[OrganizationId] = None

    connected_tenant_id: t.Optional[TenantId] = None

    metadata: t.Optional[t.Dict[str, t.Any]] = None

    onboarding_completed_at: t.Optional[datetime] = None

    pending_country: t.Optional[CountryCode] = None

    pending_email: t.Optional[str] = None
    """Email of the user being invited (express flow only)"""

    pending_organization_name: t.Optional[str] = None
    """Name of the organization to be created (express flow only)"""

    platform_customer_id: t.Optional[CustomerId] = None

    revoked_at: t.Optional[datetime] = None

File: go/connected_account.go
// this file is @generated
package meteroid

import "time"

// A connected account (relationship between platform and connected org)
type ConnectedAccount struct {
	ConnectedOrganizationId *OrganizationId `json:"connected_organization_id,omitempty"`

	ConnectedTenantId *TenantId `json:"connected_tenant_id,omitempty"`

	ConnectionType ConnectionType `json:"connection_type"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Id ConnectedAccountId `json:"id"`

	Metadata map[string]any `json:"metadata,omitempty"`

	// RFC 3339 timestamp.
	OnboardingCompletedAt *time.Time `json:"onboarding_completed_at,omitempty"`

	OnboardingMode OnboardingMode `json:"onboarding_mode"`

	PendingCountry *CountryCode `json:"pending_country,omitempty"`

	// Email of the user being invited (express flow only)
	PendingEmail *string `json:"pending_email,omitempty"`

	// Name of the organization to be created (express flow only)
	PendingOrganizationName *string `json:"pending_organization_name,omitempty"`

	PlatformCustomerId *CustomerId `json:"platform_customer_id,omitempty"`

	PlatformOrganizationId OrganizationId `json:"platform_organization_id"`

	// RFC 3339 timestamp.
	RevokedAt *time.Time `json:"revoked_at,omitempty"`

	Status ConnectionStatus `json:"status"`
}
